src/Documentation/Advanced-Concepts/Dependency-Injection.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: page
+title: Dependency Injection
+---
+
+# What is Dependency Injection
+
+Dependency injection (DI) is a software design pattern that implements inversion of control for resolving dependencies.
+
+Orleans is using the abstraction written by the developers of [ASP.NET Core](https://docs.asp.net). For a detailed explanation about how does it work, check out the [official documentation](https://docs.asp.net/en/latest/fundamentals/dependency-injection.html#dependency-injection).
+
+# DI in Orleans
+
+Dependency Injection is currently supported only on the server side within Orleans.
+
+Orleans makes it possible to inject dependencies into application [Grains](../Getting-Started-With-Orleans/Grains.md).
+
+However Orleans supports every container dependent injection mechanisms, one of the most commonly used method is constructor injection.
+
+Theoretically any type can be injected which was previously registered in a [`IServiceCollection`](https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/Extensions/DependencyInjection/IServiceCollection/index.html) during Silo startup.
+
+Orleans also makes it possible to inject the following service into an application object instance:
+- `IGrainFactory`
```

Comment:
I would remove this `IGranFactory` reference for now and we can add it later when it's actually available. Because if it doesn't make the cut for 1.3.0-beta2, then this would not be usable. And it is not usable for people in 1.2.3 yet anyway, so it is safer to just add it later